rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -4700,6 +4700,37 @@ public Unary withOperator(JLeftPadded<Type> operator) {
         }
     }
 
+    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+    @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
+    @Data
+    final class InferredType implements J, Expression, TypeTree {
```

Comment:
Not keen on adding this I think. In the case of a lambda parameter, the type expression really is not present, so I don't know that artificially creating an AST element to represent the absence of something is helpful?

Reply:
The `J.VariableDeclarations` should still have a correct type attribution.